### PR TITLE
fix(clients/java): Set a defaultRequestTimeout as 10 seconds

### LINKED ISSUE #4762: Difference between default request timeout in documentation and java client
**Description**

In the description about long polling in https://docs.zeebe.io/basics/job-workers.html#long-polling, it is stated that:

> A worker can also specify a RequestTimeout which is the duration for which the request is queued. **The default timeout is 10 seconds.**

However, `clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientBuilderImpl.java` uses 20 seconds.

This originates from a discussion in #4719

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

Backport of #5302 

## Related issues

closes #4762 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [X] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [x] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by korthout** [comment]
@saig0 can you take a look at my changes?

**Comment by korthout** [comment]
Since `0.24.3` is already frozen, I've added the notes to the `0.24.4` release notes.

File: upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java [hunk 1]
```diff
@@ -148,19 +148,25 @@
...
                               .maxJobsToActivate(1)
                               .send()
-                              .join()
-                              .getJobs()
-                              .iterator()
-                              .next();
+                              .join();
+                      assertThat(jobsResponse.getJobs()).hasSize(1);
+
+                      TestUtil.waitUntil(() -> state.hasElementInState(JOB, "ACTIVATED"));
```
Comment (saig0) [inline upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java:163]: DRY: can call `activateJob(state)` instead.

**Review by saig0 — APPROVED** [review]
Good catch :+1: 
Just one note to improve.

**Comment by aivinog1** [comment]
@korthout I see that the Jenkins build is failed, can you provide more info about the build error, please? I don't have access to the build log :)

**Comment by korthout** [comment]
@aivinog1 sorry for keeping you waiting. I've been looking into it already. It's a nasty flaky test that always seems to occur a couple test runs on this branch (but not on the PR to `0.23` and not on the PR to `develop`). I've just rebased this locally on `stable/0.24` and I'm not rerunning the build pipeline. If that helps I'd like to ask you to squash the commits and then we can merge it.

**Comment by aivinog1** [comment]
@korthout Sure, thanks, no problem 👌

**Comment by korthout** [comment]
@aivinog1 That did the trick. I've cherry-picked the changes as well to https://github.com/zeebe-io/zeebe/pull/5302 and https://github.com/zeebe-io/zeebe/pull/5319. Please squash the commits into 1 commit with a nice message and then I'll kick off the merges

**Review by korthout — APPROVED** [review]

**Comment by aivinog1** [comment]
@korthout Done, I've rebased commits in all requests :)

**Pull request merged**